werkzeug/serving.py

```diff
@@ -263,7 +263,7 @@ def version_string(self):
         return BaseHTTPRequestHandler.version_string(self).strip()
 
     def address_string(self):
-        return self.environ['REMOTE_ADDR']
+        return self.environ['REMOTE_ADDR'] if self.environ else 'localhost'
```

Comment:
You say you recieve an AttributeError, but `if self.environ` will fail the same way.